login_test.py

```diff
@@ -1,62 +1,33 @@
-from selenium import webdriver
-from selenium.webdriver.support.ui import WebDriverWait
-from selenium.webdriver.support import expected_conditions as EC
-from selenium.webdriver.common.by import By
-
-import time
 import pytest
+from fixtures import *
 
 from navigation import *
-from database_interaction import *
-from dummy_user import *
-
+from check_email import *
 
-def driver_constructor_generator():
-    driverlist = [webdriver.Firefox, webdriver.Chrome]
-    yield from driverlist
-
-def test_cantLoginWithBadCredentials():
-    global dummy_user_email
-    for driver_constructor in driver_constructor_generator():
-        with driver_constructor() as driver:
-            try_login(driver, dummy_user_email, "thatsnotmypassword")
-            assert "findes ikke" in driver.page_source
-
-def test_canLoginWithGoodCredentials(dummy_user):
-    global dummy_user_nickname
-    global dummy_user_password
-    for driver_constructor in driver_constructor_generator():
-        with driver_constructor() as driver:
-            try_login(driver, dummy_user_email, dummy_user_password)
-            assert "<title>Login</title>" not in driver.page_source
+import time
 
 
-def test_canResetPassword(dummy_user):
-    global dummy_user_email, dummy_user_password
-    for driver_constructor in driver_constructor_generator():
-        with driver_constructor() as driver:
-            navigate_to_page("login", driver)
+# Test specification number 1
+def test_unverifiedUserFirstLogin(driver, unverified_user_via_medlemshjaelp):
+    user = unverified_user_via_medlemshjaelp
 
-            forgot_password_link = driver.find_element_by_xpath("//p[@class='forgot']/a") 
-            forgot_password_link.click()
+    try_login(driver, user["email"], user["password"])
+    assert_current_page_is("bekraeft_konto", driver)
 
-            email_entry = driver.find_element_by_id("input_username")
-            email_entry.send_keys(user_email)
-            email_entry.submit()
+    mail = get_latest_mail_to(user["email"], expect_title="Aktiver din konto hos KBHFF", retry=5)
+    token = get_activation_code_from_email(mail.body)
+    fill_form_field(token, driver, form_id="input_verification_code")
+    submit_form(driver)
+    assert_current_page_is("opret_password", driver)
 
-            time.sleep(3) # wait for email to be sent
-            mail = get_latest_mail_to(dummy_user_email)
-            reset_token = get_password_reset_code_from_email(mail.body)
+    fill_form_field(user["password"], driver, form_id="input_new_password")
+    fill_form_field(user["password"], driver, form_id="input_confirm_password")
+    submit_form(driver)
+    assert_current_page_is("kvittering", driver)
 

```

Comment:
(+maybe we can include the `assert_current_page_is` at the start of each of these functions, as I guess the rest of the function is tacitly using this assumption anyway.)